src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -51,5 +76,90 @@ public void ReleaseController(object controller)
                 disposableController.Dispose();
             }
         }
+
+        /// <summary>
+        /// Activates the specified controller using the specified action context.
+        /// </summary>
+        /// <param name="controller">The controller to activate.</param>
+        /// <param name="context">The context of the executing action.</param>
+        protected virtual void Activate([NotNull] object controller, [NotNull] ActionContext context)
+        {
+            var controllerType = controller.GetType();
+            var controllerTypeInfo = controllerType.GetTypeInfo();
+            if (controllerTypeInfo.IsValueType)
+            {
+                var message = Resources.FormatValueTypesCannotBeActivated(GetType().FullName);
+                throw new InvalidOperationException(message);
+            }
+
+            var propertiesToActivate = _activateActions.GetOrAdd(controllerType,
+                                                                 _getPropertiesToActivate);
+
+            for (var i = 0; i < propertiesToActivate.Length; i++)
+            {
+                var activateInfo = propertiesToActivate[i];
+                activateInfo.Activate(controller, context);
+            }
+        }
+
+        /// <summary>
+        /// Returns a lookup of property types to delegates used to activate properties.
+        /// </summary>
+        /// <returns>The lookup.</returns>
```

Comment:
This could use some more explanation. Imagine I'm overriding this method to add my own special thing, what do I need to know?